modules/dataframe/include/inviwo/dataframe/io/csvreader.h

```diff
@@ -60,27 +47,48 @@ class IVW_MODULE_DATAFRAME_API CSVDataReaderException : public DataReaderExcepti
  */
 class IVW_MODULE_DATAFRAME_API CSVReader : public DataReaderType<DataFrame> {
 public:
-    CSVReader(const std::string& delim = ",", bool hasHeader = true, bool doubleprec = false);
+    CSVReader(std::string_view delim = defaultDelimiters, bool hasHeader = defaultFirstRowHeader,
+              bool doublePrecrecision = defaultDoublePrecision);
     CSVReader(const CSVReader&) = default;
     CSVReader(CSVReader&&) noexcept = default;
     CSVReader& operator=(const CSVReader&) = default;
     CSVReader& operator=(CSVReader&&) noexcept = default;
     virtual CSVReader* clone() const override;
     virtual ~CSVReader() = default;
 
-    void setDelimiters(const std::string& delim);
+    CSVReader& setDelimiters(const std::string& delim);
     const std::string& getDelimiters() const;
 
-    void setFirstRowHeader(bool hasHeader);
+    CSVReader& setStripQuotes(bool stripQuotes);
+    bool getStripQuotes() const;
+
+    CSVReader& setFirstRowHeader(bool hasHeader);
     bool hasFirstRowHeader() const;
 
+    CSVReader& setUnitsInHeaders(bool unitInHeaders);
+    bool hasUnitsInHeaders() const;
+
+    CSVReader& setUnitRegexp(std::string_view regexp);
+    const std::string& getUnitRegexp() const;
+
     /**
      * sets the precision for columns containing floating point values. If \p doubleprec is true,
      * values are stored as double. Otherwise float32 is used.
      */
-    void setEnableDoublePrecision(bool doubleprec);
+    CSVReader& setEnableDoublePrecision(bool doubleprec);
     bool hasDoublePrecision() const;
 
+    CSVReader& setNumberOfExampleRows(size_t rows);
+    size_t getNumberOfExamplesRows() const;
+
+    CSVReader& setLocale(std::string_view loc);
+    const std::string& getLocale() const;
+
+    enum class EmptyField { Throw, DefaultConstruct, NanOrZero };
```

Comment:
NanOrZero? I think that these options should be separated

Reply:
They already are, DefaultConstruct will generally be 0 for all types, NaNOrZero, sets NaNs for floating points and 0 for others. (for categorical ones the empty sting is not handled as a special case but just a an other value)

Reply:
Ah, I think that from a user perspective we could use Zero, or EmptyOrZero, instead of DefaultConstruct then. While DefaultConstruct might make sense from a pure programmin perspective I think the other alternatives are more informative.
```suggestion
    enum class EmptyField { Throw, 
                            EmptyOrZero, ///< "" for strings/categorical and 0 for numerical values
                            NanOrZero    ///< "" for strings/categorical, NaN for floating point, and 0 for integer values
                                            };
```